core/src/main/java/com/datastax/oss/driver/api/core/CqlSession.java

```diff
@@ -36,6 +37,18 @@ static CqlSessionBuilder builder() {
     return new CqlSessionBuilder();
   }
 
+  /**
+   * Returns a builder to create a new {@link DriverConfigLoader} that uses the default config
+   * loader as a basis, but allows providing additional configuration programmatically.
+   *
+   * <p>The built config loader may then be passed to {@link
+   * com.datastax.oss.driver.api.core.session.SessionBuilder#withConfigLoader(DriverConfigLoader)}.
+   */
+  @NonNull
+  static CqlDriverConfigLoaderBuilder configLoaderBuilder() {
```

Comment:
I think providing `configLoaderBuilder` at the `CqlSession` level is a good happy medium.  It makes it easy enough to pass in programmatic configuration, but not so obvious that it's something that users will want to by default.  So I think this should continue to encourage use of configuration files, with providing programmatic config a fallback for where that isn't viable.